packages/server/lib/plugins/child/run_require_async_child.js

```diff
@@ -113,30 +113,12 @@ function run (ipc, configFile, projectRoot) {
 
             const { devServer } = result.component
 
-            // Accounts for `devServer: require('@cypress/webpack-dev-server')
-            if (typeof devServer.devServer === 'function') {
-              on('dev-server:start', (options) => devServer.devServer(options, result.component && result.component.devServerConfig))
-
-              return setupNodeEvents(on, config)
-            }
-
             // Accounts for `devServer() {}`
             if (typeof devServer === 'function') {
```

Comment:
Is this test redundant, since we early return on `!isValidDevServer((result.component || {}))` a few lines earlier?

Reply:
My guess is the `ipc.on('setupTestingType', (testingType, options) => {` closure prevents TS from recognizing this check, and we need to do it again.